include/appfwk/DAQModule.hpp

```diff
@@ -67,17 +70,23 @@ namespace appfwk {
 class DAQModule : public NamedObject
 {
 public:
-  /**
+
+  using data_t = nlohmann::json;
```

Comment:
I might suggest "json_t" or "config_t" for clarity, if we have to have a using...

Reply:
I did also find a few places where nlohmann::json was used alongside data_t, and where data_t was used in files that didn't clearly inherit from DAQModule (for example, Logger.hpp). Maybe we could move the using to NamedObject.hpp if it's something we want to have broader visibility?

Reply:
Yes, I think some centrally defined `typedef` or `using` would be best!

As for a type name itself, it's hard for me to find a "perfect" one.  Some I've used or considered:

- `json` is what nlohmann suggests.  I am not fond because instances are not themselves really much to do with JSON per se (eg, they are not text, can be serialized to non-JSON text)
- `data_t` a bit generic even if inside the `appfwk::` namespace.
- `object_t` maybe my favorite, but instances can be arrays, numbers, strings, etc. so not totally apt
- `config_t` too specific.  instances are used also for "init", for command as a whole, for payload, for module-specific sub-payload.
- `command_t` too specific and too generic (same eg as above)
- `dynobj_t` (dynamic object) maybe apt, but a little cryptic

Maybe if no clear winner we stick with what nlohmann suggests?

BTW, `daq_application.cxx` is another spot outside of `NamedObject` that uses this type.

Once settled, I'm happy to push an update to this PR or to make fixes in a follow up.

Reply:
Right, I think there is a general agreement that a centrally defined typedef would be beneficial.
I had a look around the code to remind myself how `nlohmann::json` objects are used in this PR, and they are all tied to the concept of command. In this new scheme even initialization data is distributed via commands, therefore it seems reasonable to consider `nlohmann::json` objects as "command data" in the current context.

I would then go for `cmddata_t`, as @glehmannmiotto suggested